crates/sui-core/src/authority.rs

```diff
@@ -2002,27 +2018,24 @@ impl AuthorityState {
                     timestamp_ms,
                     tx_coins,
                     written,
-                    &module_resolver,
-                    &inner_temporary_store.loaded_runtime_objects,
+                    inner_temporary_store,
                 )
                 .await
                 .tap_ok(|_| self.metrics.post_processing_total_tx_indexed.inc())
                 .tap_err(|e| error!(?tx_digest, "Post processing - Couldn't index tx: {e}"))
                 .expect("Indexing tx should not fail");
 
             let effects: SuiTransactionBlockEffects = effects.clone().try_into()?;
+            let events = self.make_transaction_block_events(
```

Comment:
I had to put this part into its separate function because Rust won't allow me to instantiate a `dyn LayoutResolver` in a function that has enabled `trace` because it requires `Send` trait.

Reply:
I think that's fine -- it would also be fine to require that the layout_resolver API on Executor create a `Box<dyn LayoutResolver + Send>` or something (but no need to make that change now).

Reply:
Will follow up in a later PR.